sirepo/template/radia.py

```diff
@@ -1092,51 +1126,179 @@ def _save_kick_map_sdds(name, x_vals, y_vals, h_vals, v_vals, path):
+    return _update_ell_points(_update_geom_obj(o, **kwargs), beam_axis, height_axis)
+
+
+def _update_ell_points(o, beam_axis, height_axis):
+    g = _update_stemmed_points(o, beam_axis, height_axis)
+    w, h, b, c, s = g.geom
+    ax1, ax2, ay1, ay2, sx1, sx2, sy1 = g.points
+
+    o.points = _update_extrusion_points(
+        [
+            [ax1, ay1], [ax2, ay1], [ax2, ay2],
+            [sx2, ay2], [sx2, sy1], [sx1, sy1],
+            [ax1, ay1]
+        ],
+        c,
+        [int(o.stemPosition), int(o.armPosition)],
+        w,
+        b
+    )
+    return o
+
+
+def _update_jay_points(o, beam_axis, height_axis):
+    g = _update_stemmed_points(o, beam_axis, height_axis)
+    w, h, b, c, s = g.geom
+    ax1, ax2, ay1, ay2, sx1, sx2, sy1 = g.points
+    jx1 = c[0] + s[0] / 2 - o.hookWidth
+    jy1 = ay2 - o.hookHeight
+
+    o.points = _update_extrusion_points(
+        [
+            [ax1, ay1], [ax2, ay1], [ax2, jy1], [jx1, jy1], [jx1, ay2],
+            [sx2, ay2], [sx2, sy1], [sx1, sy1],
+            [ax1, ay1]
+        ],
+        c,
+        [int(o.stemPosition), int(o.armPosition)],
+        w,
+        b
+    )
+    return o
+
+
+def _update_extrusion_points(points, ctr, stem_indices, width_dir, length_dir):
+    pts = [
+        [2 * ctr[i] * stem_indices[i] + (-1)**stem_indices[i] * v for (i, v) in enumerate(p)] \
+        for p in points
     ]
-    ell.points = [[2 * c[i] * k[i] + -1**k[i] * v for (i, v) in enumerate(p)] for p in ell.points]
 
     # Radia's extrusion expects points in permutation order based on the extrusion
     # axis (x -> [y, z], y -> [z, x], z -> [x, y])
-    if w.tolist().index(1) != (b.tolist().index(1) + 1) % 3:
-        for p in ell.points:
+    if width_dir.tolist().index(1) != (length_dir.tolist().index(1) + 1) % 3:
+        for p in pts:
             p.reverse()
 
+    return pts
+
+
+def _update_geom_from_dipole(geom, dipole, beam_axis, height_axis):
+
+    width_dir, height_dir, length_dir = _geom_directions(beam_axis, height_axis)
+    _update_geom_objects(geom.objects, beam_axis, height_axis)
+
+    p = dipole.pole
+    p_obj = _find_obj_by_id(geom.objects, p.id)
+
+    if dipole.dipoleType == 'dipoleBasic':
+        pole_sz = sirepo.util.split_comma_delimited_string(p.size, float)
+        return _update_geom_obj(
+            p_obj,
+            size=pole_sz,
+            center=pole_sz * height_dir / 2 + dipole.gap * height_dir,
+            transforms=[_build_symm_xform(height_dir, _ZERO, 'parallel')]
+        )
+
+    m = dipole.magnet
+    m_obj = _find_obj_by_id(geom.objects, m.id)
+    c = dipole.coil
+    c_obj = _find_obj_by_id(geom.objects, c.id)
+    cp_obj = _find_obj_by_id(geom.objects, dipole.corePoleGroup.id)
+    mc_obj = _find_obj_by_id(geom.objects, dipole.magnetCoilGroup.id)
+
+    mag_sz = numpy.array(sirepo.util.split_comma_delimited_string(m.size, float))
+
+    if dipole.dipoleType == 'dipoleC':
+        # resize the poles to fit between the arms (minus gap)
```

Comment:
Could you use a named function instead of the comment?

Reply:
agree, see recent comment